pkgs/development/python-modules/pysam/default.nix

```diff
@@ -27,17 +28,49 @@ buildPythonPackage rec {
   };
 
   nativeBuildInputs = [ samtools ];
-  buildInputs = [ bzip2 curl cython lzma zlib ];
+  buildInputs = [
+    bzip2
+    curl
+    cython
+    libdeflate
+    lzma
+    zlib
+  ];
 
   checkInputs = [ pytest bcftools htslib ];
-  checkPhase = "py.test";
 
-  # tests require samtools<=1.9
-  doCheck = false;
-  preCheck = ''
+  # # tests require samtools<=1.9
+  # doCheck = false;
+
+  checkPhase = ''
+    # Needed to avoid /homeless-shelter error
     export HOME=$(mktemp -d)
+
+    echo "> samtools --version"
+    samtools --version
+    echo "> htsfile --version"
+    htsfile --version
+    echo "> bcftools --version"
+    bcftools --version
+
+    # Create auxiliary test data
     make -C tests/pysam_data
     make -C tests/cbcf_data
+
+    # Tests disabled due to upstream failures
+    # See https://github.com/pysam-developers/pysam/issues/961
+    py.test \
+      --deselect tests/AlignmentFileHeader_test.py::TestHeaderBAM::test_dictionary_access_works \
```

Comment:
with pytestCheck
```suggestion
  ];
    # Tests disabled due to upstream failures
    # See https://github.com/pysam-developers/pysam/issues/961
    disabledTests = [ 
      "test_dictionary_access_works"
      ...
    ];
```

Reply:
I'm using the `--deselect` approach instead of `--ignore` + `-k`. They have slightly different semantics.
The `-k` can potentially disable more tests than wanted, if another tests with the same name exists in another module/test file.

Reply:
agreed, however, pytest will print how many were deselected, and you should be able to determine that the number correlates with what you expected

Reply:
In the current package there are some of such cases, so a few passing tests are also ignored but I think this is harmless right now.

Reply:
the tests are mostly there to make sure they package is still usable.

Correctness of the package is more of a responsibility of upstream